### PR TITLE
Upgrader for version 4.0.0, set hosting goals on existing  tablets

### LINKED ISSUE #3305: OnDemand Follow-on: Set ONDEMAND hosting goal for all user table tablets during upgrade
Modify upgrade code to set ONDEMAND tablet hosting goal for all user table tablets. Remove code added as part of #3294 that handles empty `hosting:goal` column in tablet metadata. Reference comment made at: https://github.com/apache/accumulo/pull/3294#discussion_r1165817363

### PR DESCRIPTION
**Pull request by dlmarion**
Closes #3305

### CODE REVIEW

**Comment by dlmarion** [comment]
@EdColeman - this seems pretty straightforward, would be interested in your thoughts on a test.

File: server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    LOG.info("upgrade root - skipping, nothing to do");
+  }
+
+  @Override
+  public void upgradeMetadata(ServerContext context) {
+    LOG.info("upgrade metadata entries");
+    addHostingGoalToRootTable(context);
+    addHostingGoalToMetadataTable(context);
+    addHostingGoalToUserTables(context);
+  }
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java:53]: Think the following will upgrade the root tablet metadata before it loaded and the metadata table metadata before its tablets are loaded

```suggestion
  @Override
  public void upgradeZookeeper(ServerContext context) {
    LOG.info("upgrading metadata stored in zookeeper");
    addHostingGoalToRootTable(context);
  }

  @Override
  public void upgradeRoot(ServerContext context) {
    LOG.info("upgrading metadata stored in the root table");
    addHostingGoalToMetadataTable(context);
  }

  @Override
  public void upgradeMetadata(ServerContext context) {
    LOG.info("upgrade metadata stored in metadata table");
    addHostingGoalToUserTables(context);
  }
```

Comment (dlmarion) [reply server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java]: Thanks @keith-turner , that makes sense. Implemented in 911fe12.

**Review by keith-turner — APPROVED** [review]

File: server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
  */
 package org.apache.accumulo.server;
 
+import static org.apache.accumulo.server.AccumuloDataVersion.REMOVE_DEPRECATIONS_FOR_VERSION_3;
```
Comment (EdColeman) [inline server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java:21]: it looks like this is unused

**Review by EdColeman — APPROVED** [review]
If the import noted as unused is removed, then looks okay.  (surprised that the unused import is not failing a QA check)

File: server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  @Override
+  public void upgradeMetadata(ServerContext context) {
+    LOG.info("setting hosting goal on user tables");
+    addHostingGoalToUserTables(context);
+  }
+
+  private void addHostingGoalToSystemTable(ServerContext context, TableId tableId) {
+    TabletsMetadata tm =
```
Comment (EdColeman) [inline server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java:56]: TabletMetadata - should be wrapped in try-with-resources (two places in this file)

Comment (dlmarion) [reply server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java]: Addressed in 4136c63

File: server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java (hunk 2 above)
Comment (dlmarion) [reply server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java]: Addressed in 4136c63

**Review by EdColeman — APPROVED** [review]

**Pull request merged**